openstudiocore/src/model/EnergyManagementSystemMeteredOutputVariable.cpp

```diff
@@ -0,0 +1,605 @@
+    bool result = setString(OS_EnergyManagementSystem_MeteredOutputVariableFields::ResourceType, resourceType);
+    return result;
+  }
+
+  bool EnergyManagementSystemMeteredOutputVariable_Impl::setGroupType(const std::string& groupType) {
+    bool result = setString(OS_EnergyManagementSystem_MeteredOutputVariableFields::GroupType, groupType);
+    return result;
+  }
+
+  bool EnergyManagementSystemMeteredOutputVariable_Impl::setEndUseCategory(const std::string& endUseCategory) {
+    bool result = setString(OS_EnergyManagementSystem_MeteredOutputVariableFields::EndUseCategory, endUseCategory);
+    return result;
+  }
+
+  bool EnergyManagementSystemMeteredOutputVariable_Impl::setEndUseSubcategory(const std::string& endUseSubcategory) {
+    bool result = setString(OS_EnergyManagementSystem_MeteredOutputVariableFields::EndUseSubcategory, endUseSubcategory);
+    return result;
+  }
+
+  void EnergyManagementSystemMeteredOutputVariable_Impl::resetEndUseSubcategory() {
+    bool result = setString(OS_EnergyManagementSystem_MeteredOutputVariableFields::EndUseSubcategory, "");
+    OS_ASSERT(result);
+  }
+
+  bool EnergyManagementSystemMeteredOutputVariable_Impl::setUnits(const std::string& units) {
+    bool result = setString(OS_EnergyManagementSystem_MeteredOutputVariableFields::Units, units);
+    return result;
+  }
+
+  void EnergyManagementSystemMeteredOutputVariable_Impl::resetUnits() {
+    bool result = setString(OS_EnergyManagementSystem_MeteredOutputVariableFields::Units, "");
+    OS_ASSERT(result);
+  }
+
+  boost::optional<ModelObject> EnergyManagementSystemMeteredOutputVariable_Impl::eMSVariableObject() const {
+    boost::optional<ModelObject> value;
+    boost::optional<std::string> handle = getString(OS_EnergyManagementSystem_MeteredOutputVariableFields::EMSVariableName, true);
+    if (handle) {
+      const Model m = this->model();
+      value = m.getModelObject<ModelObject>(toUUID(handle.get()));
+    }
+    return value;
+  }
+
+  boost::optional< EnergyManagementSystemActuator> EnergyManagementSystemMeteredOutputVariable_Impl::emsActuator() const {
+    boost::optional<EnergyManagementSystemActuator> value;
+    boost::optional<std::string> handle = getString(OS_EnergyManagementSystem_MeteredOutputVariableFields::EMSVariableName, true);
+    if (handle) {
+      const Model m = this->model();
+      value = m.getModelObject<EnergyManagementSystemActuator>(toUUID(handle.get()));
+    }
+    return value;
+  }
+
+  boost::optional< EnergyManagementSystemSensor> EnergyManagementSystemMeteredOutputVariable_Impl::emsSensor() const {
+    boost::optional<EnergyManagementSystemSensor> value;
+    boost::optional<std::string> handle = getString(OS_EnergyManagementSystem_MeteredOutputVariableFields::EMSVariableName, true);
+    if (handle) {
+      const Model m = this->model();
+      value = m.getModelObject<EnergyManagementSystemSensor>(toUUID(handle.get()));
```

Comment:
same as the other emsVariableName or \erlVariable issue.  erl variables can be created in Programs or Subroutines (without an external object giving us a UID) so we have to treat them as strings.  However, we created object based methods to return an object if one exists (but it had to go thru the extra steps)